### DISCUSSION TITLE
Academic Conference 2024 추천 책

### DISCUSSION THREAD

**Discussion by jongfeel** [ThinkAboutSoftware/AcademicConference#329] [General]
2023 academic conference가 마지막 sprint만 남겨두고 있고 워크샵 일정도 계획되어 있습니다.

그래서 작년과 마찬가지로 미리 추천 책 목록을 지정해 두고
6월 30일 워크샵때 모인 인원이 내년에 읽을 책 세 권을 결정하는 걸로 해보도록 하겠습니다.

추천 책 목록 적는 방법은
이 글 스레드를 통해 내년에 진행했으면 하는 책 제목을 권 수와 상관 없이 올려주시면 됩니다.

작년 책 추천 글 스레드는 아래 링크를 참고하시면 됩니다.

- #196

**Comment by TaeHyoungKwon** [comment]
대략 요렇게 정해 보았습니다 🙇🏻 

나름 테마를 정해서 각 2권씩 정해보았는데요
1 -> 프로그래밍 전반적으로 이야기 하는 책을 골라 보았습니다
2 -> 이번에 했던 좋은코드 나쁜코드 처럼 실용서로 활용할 수 있는 책을 골라 보았습니다
3 -> 책이 두께가 얇지만, 생각해볼만한 내용이 많은 책을 골라 보았습니다
4 -> 팀과 팀 매니징 관련된 책을 골라 보았습니다

1. [프로그래밍 심리학](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=34228223) or [소프트웨어 장인 정신 이야기](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=312181656)
2. [객체지향의 사실과 오해](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=60550259) or [린 애자일 기법을 활용한 테스트 주도개발(ATDD)](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=40990966)
3. [대체뭐가문제야](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=22740732) or [프로그래머 철학을 만나다](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=35299409)
4. [팀토폴로지](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=258906450) or [테크니컬리더](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=22740964)

**Reply by jongfeel** [reply]
제가 안 읽어봤던 책이 두 권인데

- 린 애자일 기법을 활용한 테스트 주도 개발
- 대체 뭐가 문제야

이군요.

저는 태형님 추천 책 중에서 선택됐으면 좋은 책을 꼽아보자면 < 테크니컬리더 > 입니다.
번역본은 10년이 넘었지만, 원본은 거의 40년 전 책인데 정말 클래식하면서 좋은 영향력을 받을 수 있는 책이라고 생각하거든요.

---

신기하게 추천 책 목록에 제럴드 와인버그 책만 3권이군요 ㅎㅎ

**Reply by TaeHyoungKwon** [reply]
노린건 아닌데, 고르다보니 3개나 추가되어있긴하더라구요 ㅎㅎ

**Reply by TaeHyoungKwon** [reply]
`대체 뭐가 문제야` 책은 이전에도 한번 소개 시켜드린적이 있는데요 전 배민CEO 였던 김범준 님 EO 영상에서 개발자는 문제를 해결하는 사람이다라는 말을 하면서, 사례를 들 때 이 책의 내용을 인용합니다 - [영상](https://youtu.be/3H4umWD5bwI?t=13)

이 책은 정답을 알려주거나, 정보를 제공하는 책은 아니고 그냥 생각해볼 거리들을 많이 제공해주는 책으로 보여서 추천해보았습니다

**Comment by fkdl0048** [comment]
저는 개인적으로 읽고 싶은 책만 올리겠습니다..!

- [테스트주도개발](https://www.yes24.com/Product/Goods/12246033)

**Comment by wisdom08** [comment]
제가 읽고 싶어서 기록해뒀던 책 목록 중 같이 읽을 볼 만한 것들을 가져와봤습니다~!
- [HTTP 완벽 가이드](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=49731592)
- [프로그래머의 뇌](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=285967039) - 훌륭한 프로그래머가 알아야 할 인지과학의 모든 것
- [대체 뭐가 문제야?](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=22740732)
- [엔터프라이즈 애플리케이션 아키텍처 패턴](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=68433810) - 엔터프라이즈 애플리케이션 구축을 위한 객체지향 설계의 원리와 기법
- [구글 엔지니어는 이렇게 일한다](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=294146082) - 구글러가 전하는 문화, 프로세스, 도구의 모든 것
- [테스트 주도 개발로 배우는 객체 지향 설계와 실천](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=27490110)
- [린 애자일 기법을 활용한 테스트 주도 개발](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=40990966) - 협업을 통한 더 나은 소프트웨어 만들기
- [소프트웨어 장인](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=66925855) - 프로페셔널리즘, 실용주의, 자부심

**Reply by wisdom08** [reply]
[레거시 코드 활용 전략](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=168779479) - 손대기 두려운 낡은 코드, 안전한 변경과 테스트 기법, 재출간판

**Reply by wisdom08** [reply]
[Code Complete 코드 컴플리트 2](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=114392104) - 더 나은 소프트웨어 구현을 위한 실무 지침서

**Reply by wisdom08** [reply]
[디자인 패턴의 아름다움](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=316593468) - 객체지향 패러다임부터 설계 원칙, 코딩 규칙, 리팩터링 기법, 디자인 패턴까지

**Reply by wisdom08** [reply]
[빅데이터를 지탱하는 기술](https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=171846800) - 시시각각 변하는 데이터를 파악하는 자동화의 세계

**Comment by jongfeel** [comment]
책은 목록을 얘기해주신 태형님, 지혜님, 정안님이 책 하나씩 정하기로 했습니다

**Comment by jongfeel** [comment]
태형님: 프로그래밍 심리학
정안님: 객체지향의 사실과 오해
지혜님: 엔터프라이즈 애플리케이션 아키텍처 패턴
